### PR TITLE
WIP: Add scheme, host, and port to #req{}

### PR DESCRIPTION
**Pull request by yurrriq**

### CODE REVIEW

**Comment by yurrriq** [comment]
@tsloughter et al., what exactly do you need here?

File: src/elli_http.erl [hunk 1]
```diff
@@ -588,21 +588,33 @@ do_check_max_size_x2(_, _, _, _) -> ok.
...
-mk_req(Method, RawPath, Headers, Body, V, Socket, {Mod, Args} = Callback) ->
-    case parse_path(RawPath) of
-        {ok, {Path, URL, URLArgs}} ->
-            #req{method   = Method, path     = URL,    args    = URLArgs,
+mk_req(Method, PathTuple, Headers, Body, V, Socket, {Mod, Args} = Callback) ->
+    case parse_path(PathTuple) of
+        {ok, {SchemeHostPort, {Path, URL, URLArgs}}} ->
+            {Scheme, Host, Port} = case SchemeHostPort of
+                undefined ->
+                    list_to_tuple(lists:duplicate(3, undefined));
```
Comment (tsloughter) [inline src/elli_http.erl]: Can't `parse_path` just always return the 3-tuple so none of this has to be done. And even if for some reason we can't have `parse_path` handle that this should just be `{undefined, undefined, undefined}`.

Comment (yurrriq) [reply src/elli_http.erl]: Yep.

Comment (fenollp) [reply src/elli_http.erl]: Oh funny `-compile(inline_list_funcs).` does not inline `lists:duplicate/2`

**Comment by yurrriq** [comment]
N.B. These changes break backward compatibility for `elli_http:parse_path/1`. Should we bump to 3.1.2 then to follow semver?

**Comment by tsloughter** [comment]
To 3.0.0, I guess so.

**Comment by deadtrickster** [comment]
why was this closed?

**Comment by yurrriq** [comment]
Because I ran `git flow feature finish scheme-host-port` locally. It’s merged into develop and the release/3.0.0 branch is ready to go tonight, unless someone wants to add anything else.

**Comment by yurrriq** [comment]
Closed #28.

**Pull request closed without merging**